packages/core/src/context.js

```diff
@@ -5,15 +5,27 @@ import { useContext, forwardRef } from 'react'
 import createCache from '@emotion/cache'
 import { isBrowser } from './utils'
 
-let EmotionCacheContext: React.Context<EmotionCache | null> = React.createContext(
+const createCacheContext = () => {
+  if (isBrowser) {
+    const ssrStyles = document.querySelectorAll(`style[data-emotion]`)
+    // get SSRed styles out of the way of React's hydration
+    // document.head is a safe place to move them to
+    Array.prototype.forEach.call(ssrStyles, (node: HTMLStyleElement) => {
+      ;((document.head: any): HTMLHeadElement).appendChild(node)
+    })
```

Comment:
I think this should happen inside `@emotion/cache` since otherwise this will be broken for `emotion`/to be renamed to `@emotion/css`

Reply:
How could it affect vanilla `emotion`? This thing here only fixes moving "inline" style tags (produced by zero-config SSR) - vanilla `emotion` is not affected by this as it doesn't produce such style tags, so it should be safe to have this in `@emotion/core`

Reply:
You can use `renderStylesToString` and `renderStylesToNodeStream` with vanilla emotion which both produce inline style tags